pkg/pod/pod.go

```diff
@@ -244,8 +244,11 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1.TaskRun, taskSpec v1.Ta
 	// By default, use an empty pod template and take the one defined in the task run spec if any
 	podTemplate := pod.Template{}
 
+	// if a podTemplate is defined in the Task, pick that up here. podTemplate values from the TaskRun take precedence.
 	if taskRun.Spec.PodTemplate != nil {
 		podTemplate = *taskRun.Spec.PodTemplate
+	} else if taskSpec.PodTemplate != nil {
```

Comment:
A note here, I think this will ideally instead be a merge function that overrides only those values that the TaskRun defines instead of the whole podTemplate being replaced, but wanted to get others thoughts on this.